### PR TITLE
Host notebooks on BinderHub

### PR DESCRIPTION
**Pull request by kaedonkers**
## Problem
Currently the links in README.md only point to a static html version of the notebooks using nbviewer.

## Solution
This PR adds an `environment.yml` to allow Binder to create a Docker image from which to run the notebooks. These are fully interactive and editable so that those wishing to learn from the courses can do so without having to worry about setting up and running the notebooks on a local machine.

#### NB: Links in this PR's README.md point to `SciTools/courses/master` so will not produce a fully working notebook until this PR is merged.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: courses
```
Comment (dkillick) [inline environment.yml]: Is this the literal name of the repo? Something like `scitools-courses` might be better if this is a free text field, as that's a bit more descriptive and less likely to have a name collision!

File: README.md [hunk 2]
```diff
@@ -7,17 +7,17 @@ particularly in the fields of oceanography and meteorology.
 There are currently four courses:
 
 ### An introduction to numpy
-3.5 hours &mdash; depends on a basic Python background
-http://nbviewer.ipython.org/github/SciTools/courses/blob/master/course_content/notebooks/numpy_intro.ipynb?create=1
+3.5 hours &mdash; depends on a basic Python background
```
Comment (dkillick) [inline README.md]: Given that I've made another comment I'll just highlight this too - you've trailing whitespace at the end of each of these lines. If you need to make another code change then perhaps tidy these up too, but if not then let's get it done in a follow-up PR.

Comment (kaedonkers) [reply README.md]: The trailing double whitespace starts a [linebreak](https://gist.github.com/shaunlebron/746476e6e7a4d698b373) for the URL.\
I could use a more explicit version (`/` or `</br>`) if you would prefer.

File: environment.yml (hunk 1 above)
Comment (kaedonkers) [reply environment.yml]: I wasn't 100% sure what the name was for, but if it is the name of the environment which is created then yes `scitools-courses` makes sense!

**Comment by kaedonkers** [comment]
Thanks @dkillick!

**Comment by pelson** [comment]
Nice addition. 👍

**Pull request merged**